File: frontend/src/utils/workforceInsurance.ts
export function isMaternityLeaveInsurance(value: unknown): boolean {
  if (value == null) return false;
  const text = String(value).trim();
  if (!text) return false;
  const normalized = text.normalize('NFD').replace(/\p{M}/gu, '').toLowerCase();
  return normalized.includes('thai san');
}

File: backend/src/main/java/com/minhan/hrm/workforce/WorkforceInsurance.java
package com.minhan.hrm.workforce;

import java.text.Normalizer;
import java.util.Locale;

public final class WorkforceInsurance {

    private WorkforceInsurance() {
    }

    public static boolean isMaternityLeave(String insuranceParticipation) {
        if (insuranceParticipation == null || insuranceParticipation.isBlank()) {
            return false;
        }
        String normalized = stripAccents(insuranceParticipation).toLowerCase(Locale.ROOT);
        return normalized.contains("thai san");
    }

    private static String stripAccents(String s) {
        return Normalizer.normalize(s, Normalizer.Form.NFD).replaceAll("\\p{M}+", "");
    }
}
